File: clients/PHP-client/src/TriggersClient.php
<?php

namespace SoliDB;

class TriggersClient
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    public function list(): array
    {
        $res = $this->client->sendCommand('list_triggers', ['database' => $this->client->getDatabase()]);
        return $res['data'] ?? [];
    }

    public function listByCollection(string $collection): array
    {
        $res = $this->client->sendCommand('list_triggers_by_collection', [
            'database' => $this->client->getDatabase(),
            'collection' => $collection
        ]);
        return $res['data'] ?? [];
    }

    public function create(string $name, string $collection, string $event, string $timing, string $scriptPath, ?bool $enabled = null): array
    {
        $args = [
            'database' => $this->client->getDatabase(),
            'name' => $name,
            'collection' => $collection,
            'event' => $event,
            'timing' => $timing,
            'script_path' => $scriptPath
        ];
        if ($enabled !== null) {
            $args['enabled'] = $enabled;
        }
        $res = $this->client->sendCommand('create_trigger', $args);
        return $res['data'] ?? [];
    }

    public function get(string $triggerId): array
    {
        $res = $this->client->sendCommand('get_trigger', [
            'database' => $this->client->getDatabase(),
            'trigger_id' => $triggerId
        ]);
        return $res['data'] ?? [];
    }

    public function update(string $triggerId, array $updates): array
    {
        $res = $this->client->sendCommand('update_trigger', [
            'database' => $this->client->getDatabase(),
            'trigger_id' => $triggerId,
            'updates' => $updates
        ]);
        return $res['data'] ?? [];
    }

    public function delete(string $triggerId): void
    {
        $this->client->sendCommand('delete_trigger', [
            'database' => $this->client->getDatabase(),
            'trigger_id' => $triggerId
        ]);
    }

    public function toggle(string $triggerId, bool $enabled): void
    {
        $this->client->sendCommand('toggle_trigger', [
            'database' => $this->client->getDatabase(),
            'trigger_id' => $triggerId,
            'enabled' => $enabled
        ]);
    }
}

File: clients/Ruby-client/lib/solidb/triggers_client.rb
module SoliDB
  class TriggersClient
    def initialize(client)
      @client = client
    end

    def list
      @client.send_command("list_triggers", database: @client.database) || []
    end

    def list_by_collection(collection)
      @client.send_command("list_triggers_by_collection", database: @client.database, collection: collection) || []
    end

    def create(name:, collection:, event:, timing:, script_path:, enabled: nil)
      args = {
        database: @client.database,
        name: name,
        collection: collection,
        event: event,
        timing: timing,
        script_path: script_path
      }
      args[:enabled] = enabled unless enabled.nil?
      @client.send_command("create_trigger", args)
    end

    def get(trigger_id)
      @client.send_command("get_trigger", database: @client.database, trigger_id: trigger_id)
    end

    def update(trigger_id, updates)
      @client.send_command("update_trigger", database: @client.database, trigger_id: trigger_id, updates: updates)
    end

    def delete(trigger_id)
      @client.send_command("delete_trigger", database: @client.database, trigger_id: trigger_id)
      nil
    end

    def toggle(trigger_id, enabled)
      @client.send_command("toggle_trigger", database: @client.database, trigger_id: trigger_id, enabled: enabled)
      nil
    end
  end
end
